pkgs/development/tools/golangci-lint/default.nix

```diff
@@ -1,6 +1,6 @@
-{ buildGoModule, fetchFromGitHub, lib, installShellFiles }:
+{ buildGo123Module, fetchFromGitHub, lib, installShellFiles }:
```

Comment:
IMO this doesn't really matter

Reply:
in my local project, I override `buildGoModule` for `gopls`, `golangcl-lint` and few others, each time I update `nixpkgs`, I have to fix arguments, because `callPackage` augments keep changing

Reply:
Actually @kirillrdy is kinda right. Right now I have this in my overlay:
```
(final:prev: {
  gopls = prev.gopls.override { buildGoModule = prev.buildGo123Module; };
})
```
If this got into unstable, then my override will not work anymore. But I also think that if it's done in `all-packages.nix`, it will be hard to find out which go version is being used because `all-packages.nix` is so hard to look at in browser because of the file size.

Reply:
Having `buildGoModule_latest` or some sort like suggested by @Mic92 is the way to go IMO.

Reply:
Well it is but the tools are usually not ready for the new go version on go release date and need another update.

Reply:
Afaik go has a principle of backward compatibility and using latest go version to build packages should do no harm? Maybe I'm wrong but I never have issue building ancient packages using latest go version.

Also, we have full control on when to bump `buildGoModule_latest` up not necessarily on day one of new go release?

Reply:
For normal packages yes, go is backwards compatible, but packages that deeply go into the compile process like linters may compile but sometimes don't properly work and need a fix because some internal changed.

Also if we need to wait for a few days and updates to some packages, the point of a buildGoModule_latest twingles.

Reply:
What about something like that?

```suggestion
{ buildGo123Module
, fetchFromGitHub
, lib
, installShellFiles
, builder ? buildGo123Module
}:
```

and then users can override `builder` (or whatever we decide to name it) without worrying about it being renamed and everything remain in this file:

```
(final:prev: {
  golangci-lint = prev.golangci-lint.override { builder = prev.buildGo124Module; };
})
```

Reply:
Seems to be lots of different ways to achieve this! I want to submit a pr to do the same for [govulncheck](https://github.com/NixOS/nixpkgs/blob/5c5ec7c007298507c232e66ad45056aa7a518365/pkgs/tools/security/govulncheck/default.nix#L8) to build it with latest go for the same reason as this pr, but I'm not sure which option to choose.

Reply:
@maxbrunet that is equal to overriding buildGoModule in all-packages.